src/data/metadata.csv

```diff
@@ -600,6 +600,100 @@ AT_4,AT,Austria,4,Upper Austria,,,,,Oberösterreich,
+AT_318,AT,Austria,3,Lower Austria,318,Neunkirchen,,,,
+AT_319,AT,Austria,3,Lower Austria,319,Sankt Pölten(Land),,,,
+AT_320,AT,Austria,3,Lower Austria,320,Scheibbs,,,,
+AT_321,AT,Austria,3,Lower Austria,321,Tulln,,,,
+AT_322,AT,Austria,3,Lower Austria,322,Waidhofen an der Thaya,,,,
+AT_323,AT,Austria,3,Lower Austria,323,Wiener Neustadt(Land),,,,
+AT_325,AT,Austria,3,Lower Austria,325,Zwettl,,,,
+AT_401,AT,Austria,4,Upper Austria,401,Linz(Stadt),,,,
+AT_402,AT,Austria,4,Upper Austria,402,Steyr(Stadt),,,,
+AT_403,AT,Austria,4,Upper Austria,403,Wels(Stadt),,,,
+AT_404,AT,Austria,4,Upper Austria,404,Braunau am Inn,,,,
+AT_405,AT,Austria,4,Upper Austria,405,Eferding,,,,
+AT_406,AT,Austria,4,Upper Austria,406,Freistadt,,,,
+AT_407,AT,Austria,4,Upper Austria,407,Gmunden,,,,
+AT_408,AT,Austria,4,Upper Austria,408,Grieskirchen,,,,
+AT_409,AT,Austria,4,Upper Austria,409,Kirchdorf an der Krems,,,,
+AT_410,AT,Austria,4,Upper Austria,410,Linz-Land,,,,
+AT_411,AT,Austria,4,Upper Austria,411,Perg,,,,
+AT_412,AT,Austria,4,Upper Austria,412,Ried im Innkreis,,,,
+AT_413,AT,Austria,4,Upper Austria,413,Rohrbach,,,,
+AT_414,AT,Austria,4,Upper Austria,414,Schärding,,,,
+AT_415,AT,Austria,4,Upper Austria,415,Steyr-Land,,,,
+AT_416,AT,Austria,4,Upper Austria,416,Urfahr-Umgebung,,,,
+AT_417,AT,Austria,4,Upper Austria,417,Vöcklabruck,,,,
+AT_418,AT,Austria,4,Upper Austria,418,Wels-Land,,,,
+AT_501,AT,Austria,5,Salzburg,501,Salzburg(Stadt),,,,
+AT_502,AT,Austria,5,Salzburg,502,Hallein,,,,
+AT_503,AT,Austria,5,Salzburg,503,Salzburg-Umgebung,,,,
+AT_504,AT,Austria,5,Salzburg,504,Sankt Johann im Pongau,,,,
+AT_505,AT,Austria,5,Salzburg,505,Tamsweg,,,,
+AT_506,AT,Austria,5,Salzburg,506,Zell am See,,,,
+AT_601,AT,Austria,6,Styria,601,Graz(Stadt),,,,
+AT_603,AT,Austria,6,Styria,603,Deutschlandsberg,,,,
+AT_606,AT,Austria,6,Styria,606,Graz-Umgebung,,,,
+AT_610,AT,Austria,6,Styria,610,Leibnitz,,,,
+AT_611,AT,Austria,6,Styria,611,Leoben,,,,
+AT_612,AT,Austria,6,Styria,612,Liezen,,,,
+AT_614,AT,Austria,6,Styria,614,Murau,,,,
+AT_616,AT,Austria,6,Styria,616,Voitsberg,,,,
+AT_617,AT,Austria,6,Styria,617,Weiz,,,,
+AT_620,AT,Austria,6,Styria,620,Murtal,,,,
+AT_621,AT,Austria,6,Styria,621,Bruck-Mürzzuschlag,,,,
+AT_622,AT,Austria,6,Styria,622,Hartberg-Fürstenfeld,,,,
+AT_623,AT,Austria,6,Styria,623,Südoststeiermark,,,,
+AT_701,AT,Austria,7,Tyrol,701,Innsbruck-Stadt,,,,
+AT_702,AT,Austria,7,Tyrol,702,Imst,,,,
+AT_703,AT,Austria,7,Tyrol,703,Innsbruck-Land,,,,
+AT_704,AT,Austria,7,Tyrol,704,Kitzbühel,,,,
+AT_705,AT,Austria,7,Tyrol,705,Kufstein,,,,
+AT_706,AT,Austria,7,Tyrol,706,Landeck,,,,
+AT_707,AT,Austria,7,Tyrol,707,Lienz,,,,
+AT_708,AT,Austria,7,Tyrol,708,Reutte,,,,
+AT_709,AT,Austria,7,Tyrol,709,Schwaz,,,,
+AT_801,AT,Austria,8,Vorarlberg,801,Bludenz,,,,
+AT_802,AT,Austria,8,Vorarlberg,802,Bregenz,,,,
+AT_803,AT,Austria,8,Vorarlberg,803,Dornbirn,,,,
+AT_804,AT,Austria,8,Vorarlberg,804,Feldkirch,,,,
+AT_900,AT,Austria,9,Vienna,900,Wien,,,,
```

Comment:
It seems that `(Land)` is typically written as `-Land` ([example](https://en.wikipedia.org/wiki/Sankt_P%C3%B6lten-Land_District)) and that `(Stadt)` is omitted ([example](https://en.wikipedia.org/wiki/Linz)).

I would suggest these changes:
```suggestion
AT_101,AT,Austria,1,Burgenland,101,Eisenstadt,,,,
AT_102,AT,Austria,1,Burgenland,102,Rust,,,,
AT_103,AT,Austria,1,Burgenland,103,Eisenstadt-Umgebung,,,,
AT_104,AT,Austria,1,Burgenland,104,Güssing,,,,
AT_105,AT,Austria,1,Burgenland,105,Jennersdorf,,,,
AT_106,AT,Austria,1,Burgenland,106,Mattersburg,,,,
AT_107,AT,Austria,1,Burgenland,107,Neusiedl am See,,,,
AT_108,AT,Austria,1,Burgenland,108,Oberpullendorf,,,,
AT_109,AT,Austria,1,Burgenland,109,Oberwart,,,,
AT_201,AT,Austria,2,Carinthia,201,Klagenfurt Stadt,,,,
AT_202,AT,Austria,2,Carinthia,202,Villach Stadt,,,,
AT_203,AT,Austria,2,Carinthia,203,Hermagor,,,,
AT_204,AT,Austria,2,Carinthia,204,Klagenfurt Land,,,,
AT_205,AT,Austria,2,Carinthia,205,Sankt Veit an der Glan,,,,
AT_206,AT,Austria,2,Carinthia,206,Spittal an der Drau,,,,
AT_207,AT,Austria,2,Carinthia,207,Villach Land,,,,
AT_208,AT,Austria,2,Carinthia,208,Völkermarkt,,,,
AT_209,AT,Austria,2,Carinthia,209,Wolfsberg,,,,
AT_210,AT,Austria,2,Carinthia,210,Feldkirchen,,,,
AT_301,AT,Austria,3,Lower Austria,301,Krems an der Donau,,,,
AT_302,AT,Austria,3,Lower Austria,302,Sankt Pölten,,,,
AT_303,AT,Austria,3,Lower Austria,303,Waidhofen an der Ybbs,,,,
AT_304,AT,Austria,3,Lower Austria,304,Wiener Neustadt,,,,
AT_305,AT,Austria,3,Lower Austria,305,Amstetten,,,,
AT_306,AT,Austria,3,Lower Austria,306,Baden,,,,
AT_307,AT,Austria,3,Lower Austria,307,Bruck an der Leitha,,,,
AT_308,AT,Austria,3,Lower Austria,308,Gänserndorf,,,,
AT_309,AT,Austria,3,Lower Austria,309,Gmünd,,,,
AT_310,AT,Austria,3,Lower Austria,310,Hollabrunn,,,,
AT_311,AT,Austria,3,Lower Austria,311,Horn,,,,
AT_312,AT,Austria,3,Lower Austria,312,Korneuburg,,,,
AT_313,AT,Austria,3,Lower Austria,313,Krems-Land,,,,
AT_314,AT,Austria,3,Lower Austria,314,Lilienfeld,,,,
AT_315,AT,Austria,3,Lower Austria,315,Melk,,,,
AT_316,AT,Austria,3,Lower Austria,316,Mistelbach,,,,
AT_317,AT,Austria,3,Lower Austria,317,Mödling,,,,
AT_318,AT,Austria,3,Lower Austria,318,Neunkirchen,,,,
AT_319,AT,Austria,3,Lower Austria,319,Sankt Pölten-Land,,,,
AT_320,AT,Austria,3,Lower Austria,320,Scheibbs,,,,
AT_321,AT,Austria,3,Lower Austria,321,Tulln,,,,
AT_322,AT,Austria,3,Lower Austria,322,Waidhofen an der Thaya,,,,
AT_323,AT,Austria,3,Lower Austria,323,Wiener Neustadt-Land,,,,
AT_325,AT,Austria,3,Lower Austria,325,Zwettl,,,,
AT_401,AT,Austria,4,Upper Austria,401,Linz,,,,
AT_402,AT,Austria,4,Upper Austria,402,Steyr,,,,
AT_403,AT,Austria,4,Upper Austria,403,Wels,,,,
AT_404,AT,Austria,4,Upper Austria,404,Braunau am Inn,,,,
AT_405,AT,Austria,4,Upper Austria,405,Eferding,,,,
AT_406,AT,Austria,4,Upper Austria,406,Freistadt,,,,
AT_407,AT,Austria,4,Upper Austria,407,Gmunden,,,,
AT_408,AT,Austria,4,Upper Austria,408,Grieskirchen,,,,
AT_409,AT,Austria,4,Upper Austria,409,Kirchdorf an der Krems,,,,
AT_410,AT,Austria,4,Upper Austria,410,Linz-Land,,,,
AT_411,AT,Austria,4,Upper Austria,411,Perg,,,,
AT_412,AT,Austria,4,Upper Austria,412,Ried im Innkreis,,,,
AT_413,AT,Austria,4,Upper Austria,413,Rohrbach,,,,
AT_414,AT,Austria,4,Upper Austria,414,Schärding,,,,
AT_415,AT,Austria,4,Upper Austria,415,Steyr-Land,,,,
AT_416,AT,Austria,4,Upper Austria,416,Urfahr-Umgebung,,,,
AT_417,AT,Austria,4,Upper Austria,417,Vöcklabruck,,,,
AT_418,AT,Austria,4,Upper Austria,418,Wels-Land,,,,
AT_501,AT,Austria,5,Salzburg,501,Salzburg,,,,
AT_502,AT,Austria,5,Salzburg,502,Hallein,,,,
AT_503,AT,Austria,5,Salzburg,503,Salzburg-Umgebung,,,,
AT_504,AT,Austria,5,Salzburg,504,Sankt Johann im Pongau,,,,
AT_505,AT,Austria,5,Salzburg,505,Tamsweg,,,,
AT_506,AT,Austria,5,Salzburg,506,Zell am See,,,,
AT_601,AT,Austria,6,Styria,601,Graz,,,,
AT_603,AT,Austria,6,Styria,603,Deutschlandsberg,,,,
AT_606,AT,Austria,6,Styria,606,Graz-Umgebung,,,,
AT_610,AT,Austria,6,Styria,610,Leibnitz,,,,
AT_611,AT,Austria,6,Styria,611,Leoben,,,,
AT_612,AT,Austria,6,Styria,612,Liezen,,,,
AT_614,AT,Austria,6,Styria,614,Murau,,,,
AT_616,AT,Austria,6,Styria,616,Voitsberg,,,,
AT_617,AT,Austria,6,Styria,617,Weiz,,,,
AT_620,AT,Austria,6,Styria,620,Murtal,,,,
AT_621,AT,Austria,6,Styria,621,Bruck-Mürzzuschlag,,,,
AT_622,AT,Austria,6,Styria,622,Hartberg-Fürstenfeld,,,,
AT_623,AT,Austria,6,Styria,623,Südoststeiermark,,,,
AT_701,AT,Austria,7,Tyrol,701,Innsbruck-Stadt,,,,
AT_702,AT,Austria,7,Tyrol,702,Imst,,,,
AT_703,AT,Austria,7,Tyrol,703,Innsbruck-Land,,,,
AT_704,AT,Austria,7,Tyrol,704,Kitzbühel,,,,
AT_705,AT,Austria,7,Tyrol,705,Kufstein,,,,
AT_706,AT,Austria,7,Tyrol,706,Landeck,,,,
AT_707,AT,Austria,7,Tyrol,707,Lienz,,,,
AT_708,AT,Austria,7,Tyrol,708,Reutte,,,,
AT_709,AT,Austria,7,Tyrol,709,Schwaz,,,,
AT_801,AT,Austria,8,Vorarlberg,801,Bludenz,,,,
AT_802,AT,Austria,8,Vorarlberg,802,Bregenz,,,,
AT_803,AT,Austria,8,Vorarlberg,803,Dornbirn,,,,
AT_804,AT,Austria,8,Vorarlberg,804,Feldkirch,,,,
AT_900,AT,Austria,9,Vienna,900,Wien,,,,
```

Reply:
Where is subregion2_name actually used?  I'm pulling these names from the Austria data source (though the pipeline matches on a key constructed from the code, and not subregion2_name, so subregion2_name is irrelevant for that purpose).  Will using a more common name help in other ways, such as users searching for it?

Reply:
Names are used to identify the region in the index as you described (users searching for it). Generally, we prefer the American English names of places but it's not a strict requirement. In all cases, we want to add the keys and the corresponding Wikidata ID to the `knowledge_graph.csv` file so they can be disambiguated without depending on the name.

Reply:
> It seems that `(Land)` is typically written as `-Land` ([example](https://en.wikipedia.org/wiki/Sankt_P%C3%B6lten-Land_District)) and that `(Stadt)` is omitted ([example](https://en.wikipedia.org/wiki/Linz)).
> 
> I would suggest these changes:

Okay, I made the changes separately (whenever I said to accept suggestion it said there were recent changes ad to refresh).  FYI the changes didn't affect names that had a space before Land or a space before Stadt -- just confirming that those should stay as they are.

Also, these are not American English names I think, for example this uses Wien instead of Vienna.  Do you happen to know if these districts all have American English names?  We use American English names for the states.

Reply:
I normally get the American English names from Wikidata. The nice thing about using Wikidata as the source of truth is that you might also find locations which currently have no data, so they don't appear in the data source, but that can always change in the future.

To get a list of all municipalities in Austria from Wikidata, you'll need to:
1. Find the corresponding Wikidata property you can filter by, I normally go to the Wikipedia page of a location ([example](https://en.wikipedia.org/wiki/Sankt_P%C3%B6lten-Land_District)), then go to the Wikidata item (via the left panel, which takes me [here](https://www.wikidata.org/wiki/Q694393)) and search there for something that fits. In this case, [this](https://www.wikidata.org/wiki/Q871419) is probably what I would use. Unfortunately, I don't see the ID which you are using the code anywhere, so we might need to do manual matching.
2. Write a SPARQL script to query the Wikidata knowledge graph. There are several examples [here](https://github.com/GoogleCloudPlatform/covid-19-open-data/blob/main/src/scripts/queries.sparql) which I normally copy/paste and then change the property IDs.

Querying Wikidata and dealing with SPARQL can be pretty hard. If you get stuck, I can give it a try tomorrow and I'll attach a CSV with all the keys, names and Wikidata IDs.

Sometimes Wikidata does not have the right data (for example missing subregion codes) in which case there's nothing we can do other than manually matching each record or, if we're lucky, use string matching based on the name.

Reply:
The best idea I have is combining
https://w.wiki/pJH
and
https://w.wiki/pJK

joining the tables by wikidata id, and then manually matching with the slightly differently formatted names in our datafile.

Any thoughts on an easier way?

Reply:
Looks like the German-language query returns names which are usually just the district name preceded by "Bezirk", so I think I'll just preface the names currently in metadata.csv by "Bezirk" and try to join the tables -- that should take care of most of the cases.

Would be great if there were a way to automatically do a wikidata.org search for a list of terms and get the wikidata id of the first result.

Reply:
> Any thoughts on an easier way?

Here is what I came up with, inspired by [this](https://stackoverflow.com/a/45417484): https://w.wiki/pLG

> Would be great if there were a way to automatically do a wikidata.org search for a list of terms and get the wikidata id of the first result.

You can query one label at a time... However querying by name is very finicky, it's always much better to search based on a property. That way you also get locations which might not be captured in the data source, which can actually expose bugs in the way we process data and also guarantees that we are respecting a reasonable hierarchical order. Data sources sometimes invent new locations for the sole purpose of COVID-19 data reporting, for example.

Reply:
There are only three districts where Wikidata's names are different for EN and DE, except for the "Bezirk" prefix versus "District" suffix:
```
                                     place                 label_de   label_en
77  http://www.wikidata.org/entity/Q694399                    Krems Krems-Land
97    http://www.wikidata.org/entity/Q1741                     Wien     Vienna
100  http://www.wikidata.org/entity/Q41753 Klagenfurt am Wörthersee Klagenfurt
```

Our metadata.csv already uses `Krems-Land` and `Klagenfurt Stadt` and I suppose it makes sense to keep these and not remove the `Stadt`, correct?  Should I change `Wien` to `Vienna`?